lumigator/frontend/src/components/molecules/LExperimentTable.vue

```diff
@@ -120,28 +121,30 @@ const columnStyles = computed(() => {
   };
 });
 
-function handleRowClick(event) {
-  if (event.originalEvent.target.closest('svg.p-icon.p-datatable-row-toggle-icon')) {
+function handleRowClick(event: DataTableRowClickEvent) {
+  if (
+    (event.originalEvent.target as HTMLElement)?.closest('svg.p-icon.p-datatable-row-toggle-icon')
+  ) {
     // preventing experiment selection on row expansion
     return;
   }
   // user selected an experiment, clear selected job
-  selectedJob.value = null;
+  selectedJob.value = undefined;
   emit('l-experiment-selected', event.data);
 }
 
-function onJobSelected(job, experiment) {
+function onJobSelected(job: Job, experiment: Experiment) {
   // fetching job details from BE instead of filtering
   // because job might be still running
   experimentStore.loadJobDetails(job.id);
   // select the experiment that job belongs to
   emit('l-experiment-selected', experiment);
 }
 
-function retrieveStatus(experimentId) {
+function retrieveStatus(experimentId: string) {
   const experiment = experiments.value.find((exp) => exp.id === experimentId);
   if (!experiment) {
-    return null;
+    return undefined;
   }
```

Comment:
no need to explicitly `return undefined`, just an early `return` achieves the same result